src/PrestaShopBundle/Entity/Shipment.php

```diff
@@ -0,0 +1,291 @@
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Entity;
+
+use DateTime;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Table()
+ *
+ * @ORM\Entity()
+ *
+ * @ORM\HasLifecycleCallbacks
+ */
+class Shipment
+{
+    /**
+     * @ORM\Id
+     *
+     * @ORM\Column(name="id_shipment", type="integer")
+     *
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private int $id;
+
+    /**
+     * @ORM\Column(name="id_order", type="integer")
+     */
+    private int $orderId;
+
+    /**
+     * @ORM\Column(name="id_carrier", type="integer")
+     */
+    private int $carrierId;
+
+    /**
+     * @ORM\Column(name="id_delivery_address", type="integer")
+     */
+    private int $deliveryAddressId;
+
+    /**
+     * @ORM\Column(name="shipping_cost_tax_excl", type="float")
+     */
+    private float $shippingCostTaxExcluded;
+
+    /**
+     * @ORM\Column(name="shipping_cost_tax_incl", type="float")
+     */
+    private float $shippingCostTaxIncluded;
+
+    /**
+     * @ORM\Column(name="packed_at", type="datetime", nullable=true)
+     */
+    private ?DateTime $packedAt;
+
+    /**
+     * @ORM\Column(name="shipped_at", type="datetime", nullable=true)
+     */
+    private ?DateTime $shippedAt;
+
+    /**
+     * @ORM\Column(name="delivered_at", type="datetime", nullable=true)
+     */
+    private ?DateTime $deliveredAt;
+
+    /**
+     * @ORM\Column(name="cancelled_at", type="datetime", nullable=true)
+     */
+    private ?DateTime $cancelledAt;
+
+    /**
+     * @ORM\Column(name="tracking_number", type="string", nullable=true)
+     */
+    private ?string $trakingNumber;
+
+    /**
+     * @ORM\Column(name="date_add", type="datetime", nullable=false)
+     */
+    private DateTime $createdAt;
+
+    /**
+     * @ORM\Column(name="date_upd", type="datetime", nullable=false)
+     */
+    private DateTime $updatedAt;
+
+    /**
+     * @var Collection<ShipmentProduct>
+     *
+     * @ORM\OneToMany(targetEntity="PrestaShopBundle\Entity\ShipmentProduct", mappedBy="shipment", cascade={"persist"})
+     */
+    private Collection $products;
+
+    public function __construct()
+    {
+        $this->products = new ArrayCollection();
+    }
+
+    /**
+     * @ORM\PrePersist
+     *
+     * @ORM\PreUpdate
+     */
+    public function updatedTimestamps(): void
+    {
+        $this->updatedAt = new DateTime();
+
+        if (!isset($this->createdAt)) {
+            $this->createdAt = new DateTime();
+        }
+    }
```

Comment:
Since this is an automatic update mechanism can you place this at the end of the Entity, the idea is to focus on the getter/setters when you open/read the file as the automatic behaviours are less relevant